server-common/src/main/java/org/apache/kafka/server/common/MetadataVersion.java

```diff
@@ -186,7 +186,10 @@ public enum MetadataVersion {
     IBP_3_6_IV2(14, "3.6", "IV2", true),
 
     // Implement KIP-919 controller registration.
-    IBP_3_7_IV0(15, "3.7", "IV0", true);
+    IBP_3_7_IV0(15, "3.7", "IV0", true),
+
+    // Add ELR related supports (KIP-966).
+    IBP_3_7_IV1(16, "3.7", "IV1", true);
```

Comment:
Wouldn't having this prevent version bumps until KIP-966 is fully implemented?  Would it be enough to just have `eligible.leader.replicas.enable` config that would be used for testing, and then when the feature is ready to be release, flip `eligible.leader.replicas.enable` to `true` and add a new metadata version.

Reply:
We also need to bump the PartitionRecord version which requires a MV bump. So we have to add a new MV here. But we also have the eligible.leader.replicas.enable=false by default, so the new MV will not block other MVs.

Reply:
> Wouldn't having this prevent version bumps until KIP-966 is fully implemented 

Good question, but no, it shouldn't

One thing to note, though: if we need a second MV bump before this feature fully works, we can point that out too (and/or enforce it somehow). Do you think this will be needed @CalvinConfluent ?

Reply:
We do not need a second MV bump so far.